File: crud-item-list/src/test/java/com/gurunars/crud_item_list/PositionFetcherTest.java
package com.gurunars.crud_item_list;

import org.junit.Test;
import org.mockito.internal.util.collections.Sets;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

import static junit.framework.Assert.assertEquals;

public class PositionFetcherTest {

    private final PositionFetcher<String> positionFetcher = new PositionFetcher<>();
    private final List<String> all = Arrays.asList("one", "two", "three", "four");
    private final Set<String> selected = Sets.newSet("two", "three");

    @Test
    public void emptyAll_leadToNoPositions() throws Exception {
        assertEquals(positionFetcher.apply(new ArrayList<String>(), selected), new ArrayList<>());
    }

    @Test
    public void emptySelections_leadToNoPositions() throws Exception {
        assertEquals(positionFetcher.apply(all, new HashSet<String>()), new ArrayList<>());
    }

    @Test
    public void properSelections_leadProperPositions() throws Exception {
        assertEquals(Arrays.asList(1, 2), positionFetcher.apply(all, selected));
    }

}

File: crud-item-list/src/test/kotlin/com/gurunars/crud_item_list/PositionFetcherTest.kt
package com.gurunars.crud_item_list

import org.junit.Assert.assertEquals
import org.junit.Test


fun getRawPositions(items: List<String>, selectedItems: Set<String>): List<Int> =
    getPositions(items.itemize(), selectedItems.itemize())

class PositionFetcherTest {

    private val all = listOf("one", "two", "three", "four")
    private val selected = setOf("two", "three")

    @Test
    @Throws(Exception::class)
    fun emptyAll_leadToNoPositions() {
        assertEquals(getRawPositions(listOf(), selected), listOf<Any>())
    }

    @Test
    @Throws(Exception::class)
    fun emptySelections_leadToNoPositions() {
        assertEquals(getRawPositions(all, hashSetOf()), listOf<Any>())
    }

    @Test
    @Throws(Exception::class)
    fun properSelections_leadProperPositions() {
        assertEquals(listOf(1, 2), getRawPositions(all, selected))
    }

}
